src/renderer/hooks/usePaneNodeSearchMenu.tsx

```diff
@@ -0,0 +1,299 @@
+    readonly onConnectStop: (event: MouseEvent) => void;
+    readonly onPaneContextMenu: (event: React.MouseEvent) => void;
+}
+
+export const usePaneNodeSearchMenu = (
+    wrapperRef: React.RefObject<HTMLDivElement>
+): UsePaneNodeSearchMenuValue => {
+    const { createNode, createConnection } = useContext(GlobalVolatileContext);
+    const { closeContextMenu } = useContext(ContextMenuContext);
+    const { schemata } = useContext(GlobalContext);
+
+    const [connectingFrom, setConnectingFrom] = useState<OnConnectStartParams | null>();
+    const [connectingFromType, setConnectingFromType] = useState<string | null>();
+    const [isStoppedOnPane, setIsStoppedOnPane] = useState<boolean>(false);
+    const { getNode, project } = useReactFlow();
+
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const matchingNodes = useMemo(
+        () =>
+            getMatchingNodes(searchQuery, schemata.schemata).filter((node) => {
+                if (!connectingFrom || !connectingFromType) {
+                    return true;
+                }
+                if (connectingFrom.handleType === 'source') {
+                    return node.inputs.some((input) => {
+                        return connectingFromType === input.type && input.hasHandle;
+                    });
+                }
+                if (connectingFrom.handleType === 'target') {
+                    return node.outputs.some((output) => {
+                        return connectingFromType === output.type;
+                    });
+                }
+                log.error(`Unknown handle type: ${connectingFrom.handleType!}`);
+                return true;
+            }),
+        [connectingFrom, connectingFromType, searchQuery, schemata.schemata]
+    );
+    const byCategories = useMemo(() => getNodesByCategory(matchingNodes), [matchingNodes]);
+    const menuRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        setSearchQuery('');
+    }, [connectingFrom]);
+
+    const onPaneContextMenuNodeClick = useCallback(
+        (node: NodeSchema, position: { x: number; y: number }) => {
+            const reactFlowBounds = wrapperRef.current!.getBoundingClientRect();
+            const { x, y } = position;
+            const projPosition = project({
+                x: x - reactFlowBounds.left,
+                y: y - reactFlowBounds.top,
+            });
+            const nodeId = createUniqueId();
+            const nodeToMake: NodeProto = {
+                id: nodeId,
+                position: projPosition,
+                data: {
+                    schemaId: node.schemaId,
+                },
+                nodeType: node.nodeType,
+            };
+            createNode(nodeToMake);
+            if (isStoppedOnPane && connectingFrom) {
+                if (connectingFrom.handleType === 'source') {
+                    const firstValidHandle = schemata
+                        .get(node.schemaId)!
```

Comment:
The return value can never be null-ish. No idea why TSLint didn't catch this.
```suggestion
                        .get(node.schemaId)
```